syntax = "proto3";

package proto;

// --- Interface to push location points.

service Pusher {
  rpc PutLocation(PutLocationRequest) returns (PutLocationResponse) {}
}

message Location {
  uint64 timestamp = 1;
  uint64 user_id = 2;
  float gps_latitude = 3;
  float gps_longitude = 4;
  float gps_altitude = 5;
}

message PutLocationRequest {
  repeated Location locations = 1;
}

message PutLocationResponse {}

// --- Database internals.

message DbKey {
  int64 timestamp = 1;
  int64 user_id = 2;
  float gps_longitude_zone = 3;
  float gps_latitude_zone = 4;
}

message DbValue {
  float gps_latitude = 1;
  float gps_longitude = 2;
  float gps_altitude = 3;
}
